src/heal_location.c

```diff
@@ -35,3 +52,41 @@ const struct HealLocation *GetHealLocation(u32 index)
     else
         return &sHealLocations[index - 1];
 }
+
+bool32 IsLastHealLocation(u32 healLocation)
+{
+    const struct HealLocation *loc = GetHealLocation(healLocation);
+    const struct WarpData *warpData = &gSaveBlock1Ptr->lastHealLocation;
+
+    return warpData->mapGroup == loc->group
+        && warpData->mapNum == loc->map
+        && warpData->warpId == WARP_ID_NONE
+        && warpData->x == loc->x
+        && warpData->y == loc->y;
+}
+
+static void SetWhiteoutRespawnHealerNPCAsLastTalked(u32 healLocationId)
+{
+    gSpecialVar_LastTalked = sWhiteoutRespawnHealerNpcLocalIds[healLocationId - 1];
+}
+
+
+void SetWhiteoutRespawnWarpAndHealerNPC(struct WarpData *warp)
+{
+        u32 healLocationId = GetHealLocationIndexByWarpData(&gSaveBlock1Ptr->lastHealLocation);
+        struct HealLocation pkmCenterHealLocation = sHealLocationsPokemonCenter[healLocationId - 1];
+
+        warp->mapGroup = pkmCenterHealLocation.group;
+        warp->mapNum = pkmCenterHealLocation.map;
+        warp->warpId = WARP_ID_NONE;
+        warp->x = pkmCenterHealLocation.x;
+        warp->y = pkmCenterHealLocation.y;
+        SetWhiteoutRespawnHealerNPCAsLastTalked(healLocationId);
+}
+
+bool32 HasHealNPC(u32 healLocationId)
+{
+    if (healLocationId == HEAL_LOCATION_NONE || healLocationId >= HEAL_LOCATION_COUNT)
+        return FALSE;
+    return sWhiteoutRespawnHealerNpcLocalIds[healLocationId - 1] > 0;
```

Comment:
What is the expected behavior when no heal location is set? 

When I whiteout without ever setting a heal location, I get sent to Battle Frontier East and I heal outside lmao.

Reply:
Missing a bit of context here. What exactly isn't set? `sHealLocation`, `sHealLocationsPokemonCenter`, `sWhiteoutRespawnHealerNpcLocalIds`?

Reply:
`sHealLocation`. If you use cheat start and a script, and whiteout inside the truck, you'll see the behavior.

That's also what I was referring to with https://github.com/rh-hideout/pokeemerald-expansion/pull/4967#discussion_r1693728749

seen here: https://youtu.be/NhYclVXCmN4?t=499